### PR TITLE
Code to support fsdb

### PR DESCRIPTION
**Pull request by vishalkudale92**

### CODE REVIEW

File: cmds/fsdb.c [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+//	printf("imap.map[2] : %d \n",imap.map[2]);
+
+	lseek(devfd,((block_no*MFS_BLOCKSIZE)+offset),SEEK_SET);
+	read(devfd, (char *)uip, sizeof(struct mfs_inode));
+	return 0;
+}
+
+int read_sbnode(int block_no, int offset,struct mfs_super_block *usp)
+{
+	lseek(devfd,((block_no*MFS_BLOCKSIZE)+offset),SEEK_SET);
```
Comment (Bhushan-Jagtap-2013) [inline cmds/fsdb.c]: Please remove this comment

File: cmds/fsdb.c [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+	read(devfd, (char *)usp, sizeof(struct mfs_super_block));
+	return 0;
+}
+
+void print_sbnode(struct mfs_super_block *usp)
+{
+	printf("msb_magic                :0x%x\n",usp->msb_magic);
```
Comment (Bhushan-Jagtap-2013) [inline cmds/fsdb.c]: Please you \t instead of spaces to align text

File: cmds/fsdb.c [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+}
+
+void print_inode(struct mfs_inode *uip)
+{
+	char buf[MFS_BLOCKSIZE];
+	struct mfs_directory_entry *dirent;
+	int i,x;
+
+
+//	printf("\nInode numner : %d \n",inum);
```
Comment (Bhushan-Jagtap-2013) [inline cmds/fsdb.c]: Remove  comment

File: cmds/fsdb.c [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+	printf("mi_mode            : %x\n",uip->mi_mode);
+	printf("mi_uid             : %d\n",uip->mi_uid);
+	printf("mi_gid             : %d\n",uip->mi_gid);
```
Comment (Bhushan-Jagtap-2013) [inline cmds/fsdb.c]: use \t instead of space

File: cmds/fsdb.c [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+	printf("mi_atime           : %s\n",ctime((time_t *)&uip->mi_atime));
+	printf("mi_ctime           : %s\n",ctime((time_t *)&uip->mi_ctime));
+	printf("mi_mtime           : %s\n",ctime((time_t *)&uip->mi_mtime));
+	printf("mi_links_count     : %d\n",uip->mi_links_count);
+	printf("mi_size            : %d\n",uip->mi_size);
+	printf("mi_blocks          : %d\n",uip->mi_blocks);
+	
+	for(i=0;i<MFS_IBLOCK_COUNT;i++)
+	{
+		if(i%4==0)
```
Comment (Bhushan-Jagtap-2013) [inline cmds/fsdb.c]: I will prefer to print a table like this

block number :  1          2
address           :   4213   92340

File: cmds/fsdb.c [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+/*	if(sb.msb_magic != MFS_MAGIC)
+	{
+		printf("This is not the mfs filesystem \n");
+		exit(1);
+	}
+*/
+	block_no=atoi(argv[2]);
+	offset=atoi(argv[3]);
+
+	if(strcmp(argv[4],"inode")==0)
```
Comment (Bhushan-Jagtap-2013) [inline cmds/fsdb.c]: Please add one more entry here for directory entry 
input : address of dir entry block
output : directory entries

**Review by Bhushan-Jagtap-2013 — CHANGES_REQUESTED** [review]

File: cmds/fsdb.c (hunk 1 above)
Comment (vishalkudale92) [reply cmds/fsdb.c]: Done

File: cmds/fsdb.c (hunk 2 above)
Comment (vishalkudale92) [reply cmds/fsdb.c]: Done

File: cmds/fsdb.c (hunk 3 above)
Comment (vishalkudale92) [reply cmds/fsdb.c]: Done

File: cmds/fsdb.c (hunk 4 above)
Comment (vishalkudale92) [reply cmds/fsdb.c]: Done

File: cmds/fsdb.c (hunk 5 above)
Comment (vishalkudale92) [reply cmds/fsdb.c]: Done.

File: cmds/fsdb.c (hunk 6 above)
Comment (vishalkudale92) [reply cmds/fsdb.c]: Didn't get you. will discuss on this.

Comment (vishalkudale92) [reply cmds/fsdb.c]: Done

**Review by Bhushan-Jagtap-2013 — APPROVED** [review]

**Comment by Bhushan-Jagtap-2013** [comment]
Looks good. Need some changes but related for code formatting. We will give second round of code refactoring later.

**Pull request merged**